odoo/addons/test_new_api/tests/test_properties.py

```diff
@@ -788,6 +788,71 @@ def test_properties_field_read(self):
             msg='Definition must be present when reading child')
         self.assertEqual(many2one_property['value'], (self.partner.id, self.partner.display_name))
 
+    def test_properties_field_html(self):
+        """Test that the HTML values are sanitized."""
+        xss_payload = "<img src='x' onerror='alert(1)'/>"
+        self.message_2.attributes = [
+            {
+                "name": "html",
+                "type": "html",
+                "string": "HTML",
+                "default": xss_payload,
+                "value": xss_payload,
+                "definition_changed": True,
+            },
+        ]
+        expected = '<img src="x">'
+        self.assertEqual(dict(self.message_2.attributes)['html'], expected)
+        self.assertEqual(self.message_2.attributes['html'], expected)
+
+        properties = self.message_2.read(['attributes'])[0]['attributes']
+        self.assertEqual(properties[0]['value'], expected)
+        self.assertEqual(properties[0]['default'], expected)
+
+        definition = self.message_2.discussion.attributes_definition
+        self.assertEqual(definition[0]['default'], expected)
+
+        definition = self.message_2.discussion.read(['attributes_definition'])[0]['attributes_definition']
+        self.assertEqual(definition[0]['default'], expected)
+
+        # write a dict on the record
+        self.message_2.attributes = {'html': xss_payload}
+        self.assertEqual(self.message_2.attributes['html'], expected)
+        properties = self.message_2.read(['attributes'])[0]['attributes']
+        self.assertEqual(properties[0]['value'], expected)
+
+        # Try to trick the ORM, by first creating a text property, and then convert it to HTML
+        self.message_2.attributes = [
+            {
+                "name": "text",
+                "type": "text",
+                "string": "HTML",
+                "default": xss_payload,
+                "value": xss_payload,
+                "definition_changed": True,
+            },
+        ]
+        self.assertEqual(self.message_2.attributes['text'], xss_payload)
```

Comment:
I guess it is not a good practice to allow/encourage developers to call properties fields like this
Because after https://github.com/odoo/odoo/pull/205139. `self.message_2.attributes` returns a `Property` object whose `__getitem__` will call `convert_to_read` not in batch
`convert_to_read` will call `_get_res_ids_per_model` to check the `exists` of relevant records
if property has relational attributes, and `self.message_2.attributes['text']` is used. It would be one query per field read. The performance might be a problem in the future.